### PR TITLE
Smt12 fix index names

### PR DESCRIPTION
**Pull request by mcalmer**
Fix all the mess with index names

### CODE REVIEW

**Comment by mcalmer** [comment]
@aaannz : And here now a possible fix for the index mess

I am still checking for two places to introduce a new index. But you can start to look into these one.

**Comment by aaannz** [comment]
I don't understand the last sentence. Do you mean to do review or another way to introduce indexes?

**Comment by mcalmer** [comment]
No, I think ProductMigrations and ProductExtensions need additional indexes.
I will add them in some minutes.
So this is currently not a complete pull request.

**Comment by mcalmer** [comment]
@aaannz now this pull request is complete

File: db/schemas/mysql/3.01-3.02/110-create-drop_index_if_exists-proc.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+DROP PROCEDURE IF EXISTS smt.drop_index_if_exists;
+
+CREATE PROCEDURE smt.drop_index_if_exists(in theTable varchar(128), in theIndexName varchar(128) )
+BEGIN
+ IF((SELECT COUNT(*) AS index_exists
```
Comment (aaannz) [inline db/schemas/mysql/3.01-3.02/110-create-drop_index_if_exists-proc.sql]: This fails on schema upgrade.
From what I googled out, there is IF and IF() where it seems to me you are using syntax of IF, but calling it as IF(). Can be that the culprit?
Actual error is:
`ERROR 1064 (42000) at line 4: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '' at line 8
3.01-3.02/110-create-drop_index_if_exists-proc.sql`

**Comment by mcalmer** [comment]
@aaannz 
The reason for the error was the DELIMITER. We must change it before we create a function or procedure.
Additionally I found a problem with one index renaming.

Both is now fixed.

**Comment by aaannz** [comment]
Cool, then :+1:

**Comment by mcalmer** [comment]
I will merge this one.

**Pull request merged**